todo-model-functions.js

```diff
@@ -10,42 +10,86 @@ function Todo(title, body, priority = priorities.NO_PRIORITY, category) {
     this.category = category;
 }
 
-// todo status options
-const todoStatuses = {
-    NO: 'default',
-    PENDING: 'PENDING',
-    COMPLETED: 'COMPLETED',
-    DELETED: 'DELETETED'
-}
-
-// priority options
 const priorities = {
-    NO : 'NO_PRIORITY',
-    LOW: 'LOW_PRIORITY',
-    MEDIUM: 'MEDIUM_PRIORITY',
-    HIGH: 'HIGH_PRIORITY'
+    DEFAULT: {
+        key: 'DEFAULT',
+        value: 'No Priority',
+        order: 0,
+        indicatorHTML: '',
+        comparesTo: function(priority) {
+            return true;
+        }
+    },
+    LOW: {
+        key: 'LOW',
+        value: 'Low Priority',
+        order: 1,
+        indicatorHTML: '<span class="priority-indicator low">!</span>',
+        comparesTo: function(priority) {
+            return this.key === priority;
```

Comment:
Can we not use `this` here ?
If someone is trying to assign `LOW.comparesTo` to a variable, this'  context needs to be set